<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>Demonstration of package heritEWAS</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Demonstration of package heritEWAS</h1>



<p>DNA methylation can be thought of as a type of “mark” on DNA that can affect gene expression. Most methylation marks are erased soon after conception but methylation is known to be effectively inherited from parents to their offspring in rare cases. The <code>heritEWAS</code> package provides functions to identify these heritable DNA methylation marks using the method of (Joo et al., 2018). This approach looks for Mendelian patterns of inheritance within families and is based on the relationship structure of each family and on methylation data for a subset of the family members (but no genotype data). This method can handle large, multi-generational families and it is optimised for methylation data at thousands or millions of methylation sites, such as the data generated by epigenome-wide association studies (EWAS) of related individuals.</p>
<p>You can install the released version of heritEWAS from CRAN with:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">install.packages</span>(<span class="st">&quot;heritEWAS&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">library</span>(heritEWAS)</span></code></pre></div>
<p>To use this package, you will need two sets of data (described in more detail in the help page for the function <code>genotype_combinations</code>):</p>
<ol style="list-style-type: decimal">
<li><p>A data frame containing the pedigree data, i.e. the relationship structure of the families. Each row of the data frame corresponds to a person, and the columns correspond to each person’s individual identifier (<code>indiv</code>) and the identifiers of his or her mother (<code>mother</code>) and father (<code>father</code>), as well as a family identifier (<code>family</code>) and a binary flag (<code>typed</code>) which is <code>1</code> for people who have methylation data available. No family should contain a pedigree loop, such as one caused by inbreeding.</p></li>
<li><p>A matrix of the M-values, with rows corresponding to methylation sites (i.e. CpG probes) and columns corresponding to people. The column names should match the individual identifiers of people in the pedigree data with <code>typed = 1</code>.</p></li>
</ol>
<p>The pedigree data should look something like the following (where extra variables like <code>aff</code> and <code>age</code> can be included but they will be ignored):</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">head</span>(ped)</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="co">#&gt;   family  indiv mother father sex aff age typed</span></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="co">#&gt; 1    aey aey001   &lt;NA&gt;   &lt;NA&gt;   M   0  28     0</span></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="co">#&gt; 2    aey aey002 aey005 aey006   F   0  41     1</span></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co">#&gt; 3    aey aey003 aey002 aey001   F   0  70     1</span></span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="co">#&gt; 4    aey aey004 aey002 aey001   F   0  12     0</span></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="co">#&gt; 5    aey aey005   &lt;NA&gt;   &lt;NA&gt;   F   0  18     1</span></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="co">#&gt; 6    aey aey006 aey008 aey009   M   0  13     1</span></span>
<span id="cb3-9"><a href="#cb3-9"></a><span class="kw">unique</span>(ped<span class="op">$</span>family)</span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="co">#&gt;  [1] &quot;aey&quot; &quot;awz&quot; &quot;beu&quot; &quot;ety&quot; &quot;ibz&quot; &quot;kph&quot; &quot;ljd&quot; &quot;mps&quot; &quot;msn&quot; &quot;msp&quot; &quot;nmc&quot; &quot;nqc&quot;</span></span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="co">#&gt; [13] &quot;opq&quot; &quot;qor&quot; &quot;rpb&quot; &quot;rvn&quot; &quot;thv&quot; &quot;wkt&quot; &quot;yiu&quot; &quot;yvf&quot;</span></span></code></pre></div>
<p>And the M-values matrix should look something like:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># Colnames are the individual IDs of the pedigree data</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>M_values[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="co">#&gt;                aey002     aey003     aey005     aey006     aey007</span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co">#&gt; cg18584561 -3.8530708 -4.8507816 -3.9702189 -4.5818866 -4.2766219</span></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="co">#&gt; cg01074083 -2.3250118 -3.3565441 -0.7297221  1.3038464 -0.1940888</span></span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="co">#&gt; cg27639199 -4.0406730 -4.0770266 -3.6542961 -3.8844059 -3.6013861</span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="co">#&gt; cg26773954 -0.7413546 -1.1655354  0.9955952 -1.0797262 -0.8736215</span></span>
<span id="cb4-8"><a href="#cb4-8"></a><span class="co">#&gt; cg04417708 -0.2233104 -0.2838067  1.7158682 -0.3687787 -0.4975771</span></span></code></pre></div>
<p>The main goal of the package <code>heritEWAS</code> is to calculate a statistic <span class="math inline">\(\Delta l\)</span> for each methylation site. This statistic measures the strength of evidence that the site’s M-values follow a Mendelian pattern of inheritance within the families, with larger values of <span class="math inline">\(\Delta l\)</span> corresponding to more heritable methylation sites. This statistic is the difference in maximised log-likelihoods of two statistical models, and can be interpreted as a difference in the Bayesian information criteria of the two models; see (Joo et al. 2018).</p>
<p>The most time-consuming part of the calculation of <span class="math inline">\(\Delta l\)</span> is the same for all methylation sites, so the <code>heritEWAS</code> package calculates this part once, stores the output, then re-uses this calculation for each methylation site.<br />
This part of the calculation is performed by the function <code>genotype_combinations()</code>:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>typed_genos &lt;-<span class="st"> </span><span class="kw">genotype_combinations</span>(ped)</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="co">#&gt; Family 1 (aey): Number of genotype combinations to check: 104 </span></span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="co">#&gt; Family 2 (awz): Number of genotype combinations to check: 16 </span></span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="co">#&gt; Family 3 (beu): Number of genotype combinations to check: 96 </span></span>
<span id="cb5-5"><a href="#cb5-5"></a><span class="co">#&gt; Family 4 (ety): Number of genotype combinations to check: 66 </span></span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="co">#&gt; Family 5 (ibz): Number of genotype combinations to check: 192 </span></span>
<span id="cb5-7"><a href="#cb5-7"></a><span class="co">#&gt; Family 6 (kph): Number of genotype combinations to check: 64 </span></span>
<span id="cb5-8"><a href="#cb5-8"></a><span class="co">#&gt; Family 7 (ljd): Number of genotype combinations to check: 16 </span></span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="co">#&gt; Family 8 (mps): Number of genotype combinations to check: 68 </span></span>
<span id="cb5-10"><a href="#cb5-10"></a><span class="co">#&gt; Family 9 (msn): Number of genotype combinations to check: 66 </span></span>
<span id="cb5-11"><a href="#cb5-11"></a><span class="co">#&gt; Family 10 (msp): Number of genotype combinations to check: 132 </span></span>
<span id="cb5-12"><a href="#cb5-12"></a><span class="co">#&gt; Family 11 (nmc): Number of genotype combinations to check: 7 </span></span>
<span id="cb5-13"><a href="#cb5-13"></a><span class="co">#&gt; Family 12 (nqc): Number of genotype combinations to check: 64 </span></span>
<span id="cb5-14"><a href="#cb5-14"></a><span class="co">#&gt; Family 13 (opq): Number of genotype combinations to check: 256 </span></span>
<span id="cb5-15"><a href="#cb5-15"></a><span class="co">#&gt; Family 14 (qor): Number of genotype combinations to check: 33 </span></span>
<span id="cb5-16"><a href="#cb5-16"></a><span class="co">#&gt; Family 15 (rpb): Number of genotype combinations to check: 68 </span></span>
<span id="cb5-17"><a href="#cb5-17"></a><span class="co">#&gt; Family 16 (rvn): Number of genotype combinations to check: 16 </span></span>
<span id="cb5-18"><a href="#cb5-18"></a><span class="co">#&gt; Family 17 (thv): Number of genotype combinations to check: 16 </span></span>
<span id="cb5-19"><a href="#cb5-19"></a><span class="co">#&gt; Family 18 (wkt): Number of genotype combinations to check: 104 </span></span>
<span id="cb5-20"><a href="#cb5-20"></a><span class="co">#&gt; Family 19 (yiu): Number of genotype combinations to check: 68 </span></span>
<span id="cb5-21"><a href="#cb5-21"></a><span class="co">#&gt; Family 20 (yvf): Number of genotype combinations to check: 192</span></span></code></pre></div>
<p>The results are stored in a named list of data frames, with one data frame per family. Each data frame gives the probability of each possible combination of genotypes for those family members with methylation data (i.e. those with <code>typed = 1</code>). The possible genotypes for each person are <code>0</code> and <code>1</code>, corresponding to non-carriers and carriers (respectively) of a hypothetical genetic variant that controls methylation at a given methylation site under one of the two statistical models used to define <span class="math inline">\(\Delta l\)</span>. Impossible genotype combinations (those with a probability of <code>0</code>) are excluded from the output of <code>genotype_combinations()</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="kw">str</span>(typed_genos)</span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="co">#&gt; List of 20</span></span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="co">#&gt;  $ aey:&#39;data.frame&#39;: 36 obs. of  9 variables:</span></span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="co">#&gt;   ..$ p     : num [1:36] 0.93458 0.00935 0.00467 0.00467 0.00234 ...</span></span>
<span id="cb6-5"><a href="#cb6-5"></a><span class="co">#&gt;   ..$ aey002: num [1:36] 0 0 0 0 1 1 1 1 0 0 ...</span></span>
<span id="cb6-6"><a href="#cb6-6"></a><span class="co">#&gt;   ..$ aey003: num [1:36] 0 1 0 0 0 0 1 1 0 0 ...</span></span>
<span id="cb6-7"><a href="#cb6-7"></a><span class="co">#&gt;   ..$ aey005: num [1:36] 0 0 1 1 1 1 1 1 0 0 ...</span></span>
<span id="cb6-8"><a href="#cb6-8"></a><span class="co">#&gt;   ..$ aey006: num [1:36] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-9"><a href="#cb6-9"></a><span class="co">#&gt;   ..$ aey007: num [1:36] 0 0 0 1 0 1 0 1 0 0 ...</span></span>
<span id="cb6-10"><a href="#cb6-10"></a><span class="co">#&gt;   ..$ aey008: num [1:36] 0 0 0 0 0 0 0 0 1 1 ...</span></span>
<span id="cb6-11"><a href="#cb6-11"></a><span class="co">#&gt;   ..$ aey009: num [1:36] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-12"><a href="#cb6-12"></a><span class="co">#&gt;   ..$ aey012: num [1:36] 0 0 0 0 0 0 0 0 0 1 ...</span></span>
<span id="cb6-13"><a href="#cb6-13"></a><span class="co">#&gt;  $ awz:&#39;data.frame&#39;: 13 obs. of  5 variables:</span></span>
<span id="cb6-14"><a href="#cb6-14"></a><span class="co">#&gt;   ..$ p     : num [1:13] 0.95714 0.00476 0.00476 0.00476 0.00952 ...</span></span>
<span id="cb6-15"><a href="#cb6-15"></a><span class="co">#&gt;   ..$ awz002: int [1:13] 0 0 0 0 0 1 1 1 1 1 ...</span></span>
<span id="cb6-16"><a href="#cb6-16"></a><span class="co">#&gt;   ..$ awz007: int [1:13] 0 0 0 0 1 0 0 0 0 1 ...</span></span>
<span id="cb6-17"><a href="#cb6-17"></a><span class="co">#&gt;   ..$ awz011: int [1:13] 0 0 1 1 0 0 0 1 1 0 ...</span></span>
<span id="cb6-18"><a href="#cb6-18"></a><span class="co">#&gt;   ..$ awz013: int [1:13] 0 1 0 1 0 0 1 0 1 0 ...</span></span>
<span id="cb6-19"><a href="#cb6-19"></a><span class="co">#&gt;  $ beu:&#39;data.frame&#39;: 35 obs. of  8 variables:</span></span>
<span id="cb6-20"><a href="#cb6-20"></a><span class="co">#&gt;   ..$ p     : num [1:35] 0.93925 0.00935 0.00467 0.00234 0.00234 ...</span></span>
<span id="cb6-21"><a href="#cb6-21"></a><span class="co">#&gt;   ..$ beu001: num [1:35] 0 0 0 0 0 0 0 1 1 1 ...</span></span>
<span id="cb6-22"><a href="#cb6-22"></a><span class="co">#&gt;   ..$ beu002: num [1:35] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-23"><a href="#cb6-23"></a><span class="co">#&gt;   ..$ beu003: num [1:35] 0 0 0 0 0 0 0 0 0 1 ...</span></span>
<span id="cb6-24"><a href="#cb6-24"></a><span class="co">#&gt;   ..$ beu004: num [1:35] 0 0 0 0 0 0 0 0 1 0 ...</span></span>
<span id="cb6-25"><a href="#cb6-25"></a><span class="co">#&gt;   ..$ beu006: num [1:35] 0 0 0 1 1 1 1 0 0 0 ...</span></span>
<span id="cb6-26"><a href="#cb6-26"></a><span class="co">#&gt;   ..$ beu007: num [1:35] 0 0 1 0 0 1 1 0 0 0 ...</span></span>
<span id="cb6-27"><a href="#cb6-27"></a><span class="co">#&gt;   ..$ beu010: num [1:35] 0 1 0 0 1 0 1 0 0 0 ...</span></span>
<span id="cb6-28"><a href="#cb6-28"></a><span class="co">#&gt;  $ ety:&#39;data.frame&#39;: 51 obs. of  8 variables:</span></span>
<span id="cb6-29"><a href="#cb6-29"></a><span class="co">#&gt;   ..$ p     : num [1:51] 0.940421 0.004673 0.001168 0.000876 0.000292 ...</span></span>
<span id="cb6-30"><a href="#cb6-30"></a><span class="co">#&gt;   ..$ ety003: num [1:51] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-31"><a href="#cb6-31"></a><span class="co">#&gt;   ..$ ety004: num [1:51] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-32"><a href="#cb6-32"></a><span class="co">#&gt;   ..$ ety005: num [1:51] 0 0 0 0 0 0 0 1 1 1 ...</span></span>
<span id="cb6-33"><a href="#cb6-33"></a><span class="co">#&gt;   ..$ ety006: num [1:51] 0 0 0 1 1 1 1 0 0 1 ...</span></span>
<span id="cb6-34"><a href="#cb6-34"></a><span class="co">#&gt;   ..$ ety007: num [1:51] 0 0 1 0 0 1 1 0 1 0 ...</span></span>
<span id="cb6-35"><a href="#cb6-35"></a><span class="co">#&gt;   ..$ ety011: num [1:51] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-36"><a href="#cb6-36"></a><span class="co">#&gt;   ..$ ety013: num [1:51] 0 1 0 0 1 0 1 0 0 0 ...</span></span>
<span id="cb6-37"><a href="#cb6-37"></a><span class="co">#&gt;  $ ibz:&#39;data.frame&#39;: 99 obs. of  9 variables:</span></span>
<span id="cb6-38"><a href="#cb6-38"></a><span class="co">#&gt;   ..$ p     : num [1:99] 0.943925 0.009346 0.009346 0.000876 0.000292 ...</span></span>
<span id="cb6-39"><a href="#cb6-39"></a><span class="co">#&gt;   ..$ ibz003: num [1:99] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-40"><a href="#cb6-40"></a><span class="co">#&gt;   ..$ ibz004: num [1:99] 0 1 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-41"><a href="#cb6-41"></a><span class="co">#&gt;   ..$ ibz006: num [1:99] 0 0 0 1 1 1 1 1 1 1 ...</span></span>
<span id="cb6-42"><a href="#cb6-42"></a><span class="co">#&gt;   ..$ ibz007: num [1:99] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-43"><a href="#cb6-43"></a><span class="co">#&gt;   ..$ ibz008: num [1:99] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-44"><a href="#cb6-44"></a><span class="co">#&gt;   ..$ ibz009: num [1:99] 0 0 0 0 0 0 0 1 1 1 ...</span></span>
<span id="cb6-45"><a href="#cb6-45"></a><span class="co">#&gt;   ..$ ibz011: num [1:99] 0 0 0 0 0 1 1 0 0 1 ...</span></span>
<span id="cb6-46"><a href="#cb6-46"></a><span class="co">#&gt;   ..$ ibz016: num [1:99] 0 0 1 0 1 0 1 0 1 0 ...</span></span>
<span id="cb6-47"><a href="#cb6-47"></a><span class="co">#&gt;  $ kph:&#39;data.frame&#39;: 64 obs. of  7 variables:</span></span>
<span id="cb6-48"><a href="#cb6-48"></a><span class="co">#&gt;   ..$ p     : num [1:64] 0.94911 0.00506 0.00506 0.00506 0.00149 ...</span></span>
<span id="cb6-49"><a href="#cb6-49"></a><span class="co">#&gt;   ..$ kph001: int [1:64] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-50"><a href="#cb6-50"></a><span class="co">#&gt;   ..$ kph003: int [1:64] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-51"><a href="#cb6-51"></a><span class="co">#&gt;   ..$ kph004: int [1:64] 0 0 0 0 0 0 0 0 1 1 ...</span></span>
<span id="cb6-52"><a href="#cb6-52"></a><span class="co">#&gt;   ..$ kph005: int [1:64] 0 0 0 0 1 1 1 1 0 0 ...</span></span>
<span id="cb6-53"><a href="#cb6-53"></a><span class="co">#&gt;   ..$ kph009: int [1:64] 0 0 1 1 0 0 1 1 0 0 ...</span></span>
<span id="cb6-54"><a href="#cb6-54"></a><span class="co">#&gt;   ..$ kph011: int [1:64] 0 1 0 1 0 1 0 1 0 1 ...</span></span>
<span id="cb6-55"><a href="#cb6-55"></a><span class="co">#&gt;  $ ljd:&#39;data.frame&#39;: 13 obs. of  5 variables:</span></span>
<span id="cb6-56"><a href="#cb6-56"></a><span class="co">#&gt;   ..$ p     : num [1:13] 0.95714 0.00476 0.00476 0.00476 0.00238 ...</span></span>
<span id="cb6-57"><a href="#cb6-57"></a><span class="co">#&gt;   ..$ ljd004: int [1:13] 0 0 0 0 0 0 0 0 1 1 ...</span></span>
<span id="cb6-58"><a href="#cb6-58"></a><span class="co">#&gt;   ..$ ljd005: int [1:13] 0 0 0 0 1 1 1 1 0 1 ...</span></span>
<span id="cb6-59"><a href="#cb6-59"></a><span class="co">#&gt;   ..$ ljd009: int [1:13] 0 0 1 1 0 0 1 1 0 0 ...</span></span>
<span id="cb6-60"><a href="#cb6-60"></a><span class="co">#&gt;   ..$ ljd010: int [1:13] 0 1 0 1 0 1 0 1 0 0 ...</span></span>
<span id="cb6-61"><a href="#cb6-61"></a><span class="co">#&gt;  $ mps:&#39;data.frame&#39;: 68 obs. of  8 variables:</span></span>
<span id="cb6-62"><a href="#cb6-62"></a><span class="co">#&gt;   ..$ p     : num [1:68] 0.936186 0.009492 0.000438 0.000146 0.001606 ...</span></span>
<span id="cb6-63"><a href="#cb6-63"></a><span class="co">#&gt;   ..$ mps001: num [1:68] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-64"><a href="#cb6-64"></a><span class="co">#&gt;   ..$ mps003: num [1:68] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-65"><a href="#cb6-65"></a><span class="co">#&gt;   ..$ mps005: num [1:68] 0 0 0 0 0 0 0 0 1 1 ...</span></span>
<span id="cb6-66"><a href="#cb6-66"></a><span class="co">#&gt;   ..$ mps006: num [1:68] 0 0 0 0 1 1 1 1 0 0 ...</span></span>
<span id="cb6-67"><a href="#cb6-67"></a><span class="co">#&gt;   ..$ mps008: num [1:68] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-68"><a href="#cb6-68"></a><span class="co">#&gt;   ..$ mps011: num [1:68] 0 0 1 1 0 0 1 1 0 0 ...</span></span>
<span id="cb6-69"><a href="#cb6-69"></a><span class="co">#&gt;   ..$ mps016: num [1:68] 0 1 0 1 0 1 0 1 0 1 ...</span></span>
<span id="cb6-70"><a href="#cb6-70"></a><span class="co">#&gt;  $ msn:&#39;data.frame&#39;: 50 obs. of  8 variables:</span></span>
<span id="cb6-71"><a href="#cb6-71"></a><span class="co">#&gt;   ..$ p     : num [1:50] 0.936332 0.00993 0.000876 0.000876 0.000292 ...</span></span>
<span id="cb6-72"><a href="#cb6-72"></a><span class="co">#&gt;   ..$ msn001: num [1:50] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-73"><a href="#cb6-73"></a><span class="co">#&gt;   ..$ msn003: num [1:50] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-74"><a href="#cb6-74"></a><span class="co">#&gt;   ..$ msn005: num [1:50] 0 0 0 0 0 0 1 1 1 1 ...</span></span>
<span id="cb6-75"><a href="#cb6-75"></a><span class="co">#&gt;   ..$ msn007: num [1:50] 0 0 1 1 1 1 0 0 1 1 ...</span></span>
<span id="cb6-76"><a href="#cb6-76"></a><span class="co">#&gt;   ..$ msn010: num [1:50] 0 1 0 0 1 1 0 1 0 0 ...</span></span>
<span id="cb6-77"><a href="#cb6-77"></a><span class="co">#&gt;   ..$ msn011: num [1:50] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-78"><a href="#cb6-78"></a><span class="co">#&gt;   ..$ msn013: num [1:50] 0 0 0 1 0 1 0 0 0 1 ...</span></span>
<span id="cb6-79"><a href="#cb6-79"></a><span class="co">#&gt;  $ msp:&#39;data.frame&#39;: 132 obs. of  9 variables:</span></span>
<span id="cb6-80"><a href="#cb6-80"></a><span class="co">#&gt;   ..$ p     : num [1:132] 0.931732 0.000365 0.000365 0.000365 0.001825 ...</span></span>
<span id="cb6-81"><a href="#cb6-81"></a><span class="co">#&gt;   ..$ msp001: num [1:132] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-82"><a href="#cb6-82"></a><span class="co">#&gt;   ..$ msp004: num [1:132] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-83"><a href="#cb6-83"></a><span class="co">#&gt;   ..$ msp008: num [1:132] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-84"><a href="#cb6-84"></a><span class="co">#&gt;   ..$ msp009: num [1:132] 0 0 0 0 0 0 0 0 1 1 ...</span></span>
<span id="cb6-85"><a href="#cb6-85"></a><span class="co">#&gt;   ..$ msp010: num [1:132] 0 0 0 0 1 1 1 1 0 0 ...</span></span>
<span id="cb6-86"><a href="#cb6-86"></a><span class="co">#&gt;   ..$ msp011: num [1:132] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-87"><a href="#cb6-87"></a><span class="co">#&gt;   ..$ msp013: int [1:132] 0 0 1 1 0 0 1 1 0 0 ...</span></span>
<span id="cb6-88"><a href="#cb6-88"></a><span class="co">#&gt;   ..$ msp014: int [1:132] 0 1 0 1 0 1 0 1 0 1 ...</span></span>
<span id="cb6-89"><a href="#cb6-89"></a><span class="co">#&gt;  $ nmc:&#39;data.frame&#39;: 6 obs. of  5 variables:</span></span>
<span id="cb6-90"><a href="#cb6-90"></a><span class="co">#&gt;   ..$ p     : num [1:6] 0.94286 0.00952 0.00952 0.00952 0.00952 ...</span></span>
<span id="cb6-91"><a href="#cb6-91"></a><span class="co">#&gt;   ..$ nmc001: num [1:6] 0 0 1 0 1 0</span></span>
<span id="cb6-92"><a href="#cb6-92"></a><span class="co">#&gt;   ..$ nmc002: num [1:6] 0 1 1 0 0 0</span></span>
<span id="cb6-93"><a href="#cb6-93"></a><span class="co">#&gt;   ..$ nmc003: num [1:6] 0 0 0 1 1 0</span></span>
<span id="cb6-94"><a href="#cb6-94"></a><span class="co">#&gt;   ..$ nmc010: num [1:6] 0 0 0 0 0 1</span></span>
<span id="cb6-95"><a href="#cb6-95"></a><span class="co">#&gt;  $ nqc:&#39;data.frame&#39;: 49 obs. of  7 variables:</span></span>
<span id="cb6-96"><a href="#cb6-96"></a><span class="co">#&gt;   ..$ p     : num [1:49] 0.94217 0.00292 0.00292 0.00292 0.00292 ...</span></span>
<span id="cb6-97"><a href="#cb6-97"></a><span class="co">#&gt;   ..$ nqc004: int [1:49] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-98"><a href="#cb6-98"></a><span class="co">#&gt;   ..$ nqc005: int [1:49] 0 0 0 0 0 0 0 0 0 1 ...</span></span>
<span id="cb6-99"><a href="#cb6-99"></a><span class="co">#&gt;   ..$ nqc008: int [1:49] 0 0 0 0 0 0 0 0 1 0 ...</span></span>
<span id="cb6-100"><a href="#cb6-100"></a><span class="co">#&gt;   ..$ nqc009: int [1:49] 0 0 0 0 1 1 1 1 0 0 ...</span></span>
<span id="cb6-101"><a href="#cb6-101"></a><span class="co">#&gt;   ..$ nqc012: int [1:49] 0 0 1 1 0 0 1 1 0 0 ...</span></span>
<span id="cb6-102"><a href="#cb6-102"></a><span class="co">#&gt;   ..$ nqc013: int [1:49] 0 1 0 1 0 1 0 1 0 0 ...</span></span>
<span id="cb6-103"><a href="#cb6-103"></a><span class="co">#&gt;  $ opq:&#39;data.frame&#39;: 151 obs. of  9 variables:</span></span>
<span id="cb6-104"><a href="#cb6-104"></a><span class="co">#&gt;   ..$ p     : num [1:151] 0.93925 0.00117 0.00117 0.00117 0.00117 ...</span></span>
<span id="cb6-105"><a href="#cb6-105"></a><span class="co">#&gt;   ..$ opq002: int [1:151] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-106"><a href="#cb6-106"></a><span class="co">#&gt;   ..$ opq004: int [1:151] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-107"><a href="#cb6-107"></a><span class="co">#&gt;   ..$ opq005: int [1:151] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-108"><a href="#cb6-108"></a><span class="co">#&gt;   ..$ opq008: int [1:151] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-109"><a href="#cb6-109"></a><span class="co">#&gt;   ..$ opq011: int [1:151] 0 0 0 0 0 0 0 0 1 1 ...</span></span>
<span id="cb6-110"><a href="#cb6-110"></a><span class="co">#&gt;   ..$ opq013: int [1:151] 0 0 0 0 1 1 1 1 0 0 ...</span></span>
<span id="cb6-111"><a href="#cb6-111"></a><span class="co">#&gt;   ..$ opq014: int [1:151] 0 0 1 1 0 0 1 1 0 0 ...</span></span>
<span id="cb6-112"><a href="#cb6-112"></a><span class="co">#&gt;   ..$ opq015: int [1:151] 0 1 0 1 0 1 0 1 0 1 ...</span></span>
<span id="cb6-113"><a href="#cb6-113"></a><span class="co">#&gt;  $ qor:&#39;data.frame&#39;: 26 obs. of  7 variables:</span></span>
<span id="cb6-114"><a href="#cb6-114"></a><span class="co">#&gt;   ..$ p     : num [1:26] 0.92775 0.00573 0.00573 0.00573 0.00573 ...</span></span>
<span id="cb6-115"><a href="#cb6-115"></a><span class="co">#&gt;   ..$ qor002: num [1:26] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-116"><a href="#cb6-116"></a><span class="co">#&gt;   ..$ qor005: num [1:26] 0 0 0 0 0 0 0 0 1 1 ...</span></span>
<span id="cb6-117"><a href="#cb6-117"></a><span class="co">#&gt;   ..$ qor010: num [1:26] 0 0 0 0 1 1 1 1 0 0 ...</span></span>
<span id="cb6-118"><a href="#cb6-118"></a><span class="co">#&gt;   ..$ qor011: num [1:26] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-119"><a href="#cb6-119"></a><span class="co">#&gt;   ..$ qor014: num [1:26] 0 0 1 1 0 0 1 1 0 0 ...</span></span>
<span id="cb6-120"><a href="#cb6-120"></a><span class="co">#&gt;   ..$ qor017: num [1:26] 0 1 0 1 0 1 0 1 0 1 ...</span></span>
<span id="cb6-121"><a href="#cb6-121"></a><span class="co">#&gt;  $ rpb:&#39;data.frame&#39;: 44 obs. of  8 variables:</span></span>
<span id="cb6-122"><a href="#cb6-122"></a><span class="co">#&gt;   ..$ p     : num [1:44] 0.945238 0.000595 0.000595 0.000595 0.000595 ...</span></span>
<span id="cb6-123"><a href="#cb6-123"></a><span class="co">#&gt;   ..$ rpb001: num [1:44] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-124"><a href="#cb6-124"></a><span class="co">#&gt;   ..$ rpb003: num [1:44] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-125"><a href="#cb6-125"></a><span class="co">#&gt;   ..$ rpb004: num [1:44] 0 0 0 0 0 1 1 1 1 1 ...</span></span>
<span id="cb6-126"><a href="#cb6-126"></a><span class="co">#&gt;   ..$ rpb006: num [1:44] 0 1 1 1 1 0 1 1 1 1 ...</span></span>
<span id="cb6-127"><a href="#cb6-127"></a><span class="co">#&gt;   ..$ rpb007: num [1:44] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-128"><a href="#cb6-128"></a><span class="co">#&gt;   ..$ rpb009: int [1:44] 0 0 0 1 1 0 0 0 1 1 ...</span></span>
<span id="cb6-129"><a href="#cb6-129"></a><span class="co">#&gt;   ..$ rpb014: int [1:44] 0 0 1 0 1 0 0 1 0 1 ...</span></span>
<span id="cb6-130"><a href="#cb6-130"></a><span class="co">#&gt;  $ rvn:&#39;data.frame&#39;: 13 obs. of  5 variables:</span></span>
<span id="cb6-131"><a href="#cb6-131"></a><span class="co">#&gt;   ..$ p     : num [1:13] 0.95714 0.00476 0.00476 0.00476 0.00952 ...</span></span>
<span id="cb6-132"><a href="#cb6-132"></a><span class="co">#&gt;   ..$ rvn005: int [1:13] 0 0 0 0 0 1 1 1 1 1 ...</span></span>
<span id="cb6-133"><a href="#cb6-133"></a><span class="co">#&gt;   ..$ rvn007: int [1:13] 0 0 0 0 1 0 0 0 0 1 ...</span></span>
<span id="cb6-134"><a href="#cb6-134"></a><span class="co">#&gt;   ..$ rvn009: int [1:13] 0 0 1 1 0 0 0 1 1 0 ...</span></span>
<span id="cb6-135"><a href="#cb6-135"></a><span class="co">#&gt;   ..$ rvn013: int [1:13] 0 1 0 1 0 0 1 0 1 0 ...</span></span>
<span id="cb6-136"><a href="#cb6-136"></a><span class="co">#&gt;  $ thv:&#39;data.frame&#39;: 16 obs. of  5 variables:</span></span>
<span id="cb6-137"><a href="#cb6-137"></a><span class="co">#&gt;   ..$ p     : num [1:16] 0.95595 0.00357 0.00357 0.00357 0.00357 ...</span></span>
<span id="cb6-138"><a href="#cb6-138"></a><span class="co">#&gt;   ..$ thv005: int [1:16] 0 0 0 0 0 0 0 0 1 1 ...</span></span>
<span id="cb6-139"><a href="#cb6-139"></a><span class="co">#&gt;   ..$ thv006: int [1:16] 0 0 0 0 1 1 1 1 0 0 ...</span></span>
<span id="cb6-140"><a href="#cb6-140"></a><span class="co">#&gt;   ..$ thv011: int [1:16] 0 0 1 1 0 0 1 1 0 0 ...</span></span>
<span id="cb6-141"><a href="#cb6-141"></a><span class="co">#&gt;   ..$ thv012: int [1:16] 0 1 0 1 0 1 0 1 0 1 ...</span></span>
<span id="cb6-142"><a href="#cb6-142"></a><span class="co">#&gt;  $ wkt:&#39;data.frame&#39;: 45 obs. of  9 variables:</span></span>
<span id="cb6-143"><a href="#cb6-143"></a><span class="co">#&gt;   ..$ p     : num [1:45] 0.942857 0.004762 0.004762 0.000595 0.000595 ...</span></span>
<span id="cb6-144"><a href="#cb6-144"></a><span class="co">#&gt;   ..$ wkt004: num [1:45] 0 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb6-145"><a href="#cb6-145"></a><span class="co">#&gt;   ..$ wkt007: num [1:45] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-146"><a href="#cb6-146"></a><span class="co">#&gt;   ..$ wkt008: num [1:45] 0 0 0 1 1 1 1 1 1 1 ...</span></span>
<span id="cb6-147"><a href="#cb6-147"></a><span class="co">#&gt;   ..$ wkt011: num [1:45] 0 0 1 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-148"><a href="#cb6-148"></a><span class="co">#&gt;   ..$ wkt012: num [1:45] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-149"><a href="#cb6-149"></a><span class="co">#&gt;   ..$ wkt013: num [1:45] 0 0 0 0 0 0 0 1 1 1 ...</span></span>
<span id="cb6-150"><a href="#cb6-150"></a><span class="co">#&gt;   ..$ wkt014: num [1:45] 0 0 0 0 0 1 1 0 0 1 ...</span></span>
<span id="cb6-151"><a href="#cb6-151"></a><span class="co">#&gt;   ..$ wkt015: num [1:45] 0 0 0 0 1 0 1 0 1 0 ...</span></span>
<span id="cb6-152"><a href="#cb6-152"></a><span class="co">#&gt;  $ yiu:&#39;data.frame&#39;: 47 obs. of  8 variables:</span></span>
<span id="cb6-153"><a href="#cb6-153"></a><span class="co">#&gt;   ..$ p     : num [1:47] 0.93575 0.00117 0.00117 0.00117 0.00584 ...</span></span>
<span id="cb6-154"><a href="#cb6-154"></a><span class="co">#&gt;   ..$ yiu001: num [1:47] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-155"><a href="#cb6-155"></a><span class="co">#&gt;   ..$ yiu003: num [1:47] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-156"><a href="#cb6-156"></a><span class="co">#&gt;   ..$ yiu005: num [1:47] 0 0 0 0 0 0 0 0 1 1 ...</span></span>
<span id="cb6-157"><a href="#cb6-157"></a><span class="co">#&gt;   ..$ yiu008: num [1:47] 0 0 0 0 1 1 1 1 0 0 ...</span></span>
<span id="cb6-158"><a href="#cb6-158"></a><span class="co">#&gt;   ..$ yiu011: num [1:47] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-159"><a href="#cb6-159"></a><span class="co">#&gt;   ..$ yiu014: int [1:47] 0 0 1 1 0 0 1 1 0 0 ...</span></span>
<span id="cb6-160"><a href="#cb6-160"></a><span class="co">#&gt;   ..$ yiu018: int [1:47] 0 1 0 1 0 1 0 1 0 1 ...</span></span>
<span id="cb6-161"><a href="#cb6-161"></a><span class="co">#&gt;  $ yvf:&#39;data.frame&#39;: 63 obs. of  9 variables:</span></span>
<span id="cb6-162"><a href="#cb6-162"></a><span class="co">#&gt;   ..$ p     : num [1:63] 0.93224 0.00234 0.00234 0.00234 0.00234 ...</span></span>
<span id="cb6-163"><a href="#cb6-163"></a><span class="co">#&gt;   ..$ yvf001: num [1:63] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-164"><a href="#cb6-164"></a><span class="co">#&gt;   ..$ yvf004: num [1:63] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-165"><a href="#cb6-165"></a><span class="co">#&gt;   ..$ yvf006: num [1:63] 0 0 0 0 0 0 0 0 1 1 ...</span></span>
<span id="cb6-166"><a href="#cb6-166"></a><span class="co">#&gt;   ..$ yvf007: num [1:63] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-167"><a href="#cb6-167"></a><span class="co">#&gt;   ..$ yvf008: num [1:63] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb6-168"><a href="#cb6-168"></a><span class="co">#&gt;   ..$ yvf011: int [1:63] 0 0 0 0 1 1 1 1 0 0 ...</span></span>
<span id="cb6-169"><a href="#cb6-169"></a><span class="co">#&gt;   ..$ yvf013: int [1:63] 0 0 1 1 0 0 1 1 0 0 ...</span></span>
<span id="cb6-170"><a href="#cb6-170"></a><span class="co">#&gt;   ..$ yvf014: int [1:63] 0 1 0 1 0 1 0 1 0 1 ...</span></span></code></pre></div>
<p>Given the genotype probabilities, we can use the package’s main function <code>ML_estimates()</code> to compute <span class="math inline">\(\Delta l\)</span> for each site. The output is a data frame with rows corresponding to methylation sites and columns giving details about certain fitted models (see the help page of the function <code>ML_estimates</code> for more details). In particular, the column <code>delta.l</code> gives the statistic <span class="math inline">\(\Delta l\)</span>, which measures how heritable each methylation site is.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>MLEs &lt;-<span class="st"> </span><span class="kw">ML_estimates</span>(typed_genos, M_values, <span class="dt">ncores =</span> <span class="dv">2</span>)</span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="co">#&gt; Loading data</span></span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="co">#&gt; Computing log-likelihood for the standard mixture model</span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co">#&gt; Computing log-likelihood for the Mendelian model</span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="kw">head</span>(MLEs)</span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="co">#&gt;     methylation.site   delta.l    ll.null    ll.mix ll.mendel    mu.null</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co">#&gt; 3         cg27639199 118.09775 -271.71287 -264.2143 -146.1166 -2.6183850</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="co">#&gt; 1         cg18584561  90.00927 -237.56529 -231.8907 -141.8814 -3.2970715</span></span>
<span id="cb7-9"><a href="#cb7-9"></a><span class="co">#&gt; 5         cg04417708  54.40022 -249.14781 -241.2967 -186.8965  0.5123842</span></span>
<span id="cb7-10"><a href="#cb7-10"></a><span class="co">#&gt; 486       cg20592836  49.01353 -287.01184 -275.2788 -226.2653 -2.2974237</span></span>
<span id="cb7-11"><a href="#cb7-11"></a><span class="co">#&gt; 380       cg06536614  47.62643   48.11356  103.9804  151.6069 -2.8489881</span></span>
<span id="cb7-12"><a href="#cb7-12"></a><span class="co">#&gt; 6         cg05865327  26.85557 -262.41616 -257.5835 -230.7279  0.2555829</span></span>
<span id="cb7-13"><a href="#cb7-13"></a><span class="co">#&gt;       sd.null mu0.mendel sd0.mendel mu1.mendel sd1.mendel    mu0.mix    sd0.mix</span></span>
<span id="cb7-14"><a href="#cb7-14"></a><span class="co">#&gt; 3   2.0214565 -3.7764376 0.21607159  0.4513136  1.3477016 -2.5365336 2.06441478</span></span>
<span id="cb7-15"><a href="#cb7-15"></a><span class="co">#&gt; 1   1.5481184 -4.2069460 0.25838259 -0.8795158  0.7369130 -3.2561007 1.56614986</span></span>
<span id="cb7-16"><a href="#cb7-16"></a><span class="co">#&gt; 5   1.6947389 -0.4606702 0.68962203  3.0981023  0.1964459  0.3379929 1.60045489</span></span>
<span id="cb7-17"><a href="#cb7-17"></a><span class="co">#&gt; 486 2.2780993 -1.1515825 2.30146074 -4.0355995  0.1204863 -2.1302684 2.31615821</span></span>
<span id="cb7-18"><a href="#cb7-18"></a><span class="co">#&gt; 380 0.1661564 -2.9096112 0.02112367 -2.6747390  0.2544657 -2.9061301 0.05273047</span></span>
<span id="cb7-19"><a href="#cb7-19"></a><span class="co">#&gt; 6   1.8798420 -0.5085749 1.44145564  2.7821332  0.1835955  0.1361195 1.83725157</span></span>
<span id="cb7-20"><a href="#cb7-20"></a><span class="co">#&gt;       mu1.mix    sd1.mix</span></span>
<span id="cb7-21"><a href="#cb7-21"></a><span class="co">#&gt; 3   -3.811011 0.09404426</span></span>
<span id="cb7-22"><a href="#cb7-22"></a><span class="co">#&gt; 1   -4.292643 0.01360576</span></span>
<span id="cb7-23"><a href="#cb7-23"></a><span class="co">#&gt; 5    3.190368 0.06669716</span></span>
<span id="cb7-24"><a href="#cb7-24"></a><span class="co">#&gt; 486 -4.047780 0.04947619</span></span>
<span id="cb7-25"><a href="#cb7-25"></a><span class="co">#&gt; 380 -2.453558 0.14302755</span></span>
<span id="cb7-26"><a href="#cb7-26"></a><span class="co">#&gt; 6    2.828646 0.12484766</span></span></code></pre></div>
<div id="references" class="section level2">
<h2>References</h2>
<p>Joo JE, Dowty JG, Milne RL, Wong EM, Dugue PA, English D, Hopper JL, Goldgar DE, Giles GG, Southey MC, kConFab. Heritable DNA methylation marks associated with susceptibility to breast cancer. Nat Commun. 2018 Feb 28;9(1):867. </p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
